### ISSUE TITLE
When searching by file, automatically switch to that file's diffs when switching commits

### ISSUE THREAD

**Issue by jetmore** [sublimehq/sublime_merge#931]
**Problem description**

I want to only look at changes for a specific file in my current branch.  I filter by file:path/to/file.ext.  I select a commit, then I select file.ext from the list of diffs.  When I click on a different commit, it resets the diff pane back to the summary.

**Preferred solution**

I would like it if the diff pane for file.ext (the file I'm filtered on) stayed active as I switched between commits

**Alternatives**

None that I'm aware of.  Moving the files and the commits to different tabs helps some, but not if there are a large amount of files in the commit

Thanks for your consideration

**Comment by dpjohnst** [comment]
Hi @jetmore,

Thank you for sharing this feedback with us!

Just to clarify, when you select a file diff from the tabs (or files list), you'd like that selection to be preserved when you change between commits?

Kind regards,
\- Dylan

**Comment by jetmore** [comment]
Yes, that's it exactly.  It's not terrible to reselect the file from the files lists (or tabs) when there aren't that many files, but when there are 10s or 100s of files in a commit, it really slows things down to find it and select it every time a new commit is selected.

Here's my specific use case, in case there's a better way to do this. I'm rebasing a feature branch with roughly 60 commits in it.  I have one commit which is a one-line fix to an earlier commit.  I want to squash it into the commit which it's fixing. Because this file has been changed many times in many ways in this feature branch, I want to search for the file (so I only have commits in which the file was changed) and then quickly click through the commits in which the file was changed to find the most applicable commit to merge this fix into. Having to re-find the file every time I click a new commit takes more time than the actual file evaluation does.

Thanks
--John

**Comment by srbs** [comment]
@dpjohnst, SM has a few different view-preservation issues when switching commits.

* file tab (this issue)
* blame scroll position (#626)
* file tree opened/closed folders (can't seem to find the issue or forum post)

If all three are fixed, I think the commit history usability of SM will go up quite drastically.
